### ISSUE TITLE
docs: document bug in safari that caches css and prevents hmr

### ISSUE THREAD

**Issue by powersurge360** [ElMassimo/vite_ruby#258]
[troubleshooting section]: https://vite-ruby.netlify.app/guide/troubleshooting.html

- [X] I have tried upgrading by running `bundle update vite_ruby`.
- [X] I have read the __[troubleshooting section]__ before opening an issue.

### Description 📖

In Safari on macOS (at least) when using `vite_stylesheet_tag` it is difficult to get new copies of a stylesheet as it changes. I believe this is because Safari isn't respecting the `Cache-Control: no-cache` when an `ETag` header is set. I am unsure if this is a bug in rails, vite-ruby or vite-rails. I was unable to work out where the `ETag` is set. I also tried in Chrome and there didn't seem to be an issue.

### Reproduction 🐞

_Please provide a link to a repo that can reproduce the problem you ran into._

<details>
<summary>Vite Ruby Info</summary>

_Run `bin/rake vite:info` and provide the output:_

```

```
</details>

### Logs 📜

_If not providing a reproduction:_

<details>
<summary>Output</summary>

_Run `DEBUG=vite-plugin-ruby:* bin/vite dev` or `DEBUG=vite-plugin-ruby:* bin/vite build` and provide the output:_

```
bin/vite present?: true
vite_ruby: 3.2.3
vite_rails: 3.0.12
rails: 7.0.3.1
node: v16.17.0
npm: 8.15.0
yarn: 1.22.19
pnpm: 7.5.2
ruby: ruby 3.1.2p20 (2022-04-12 revision 4491bb740a) [arm64-darwin21]

installed packages:
pimiibo@ /Users/powersurge360/projects/pimiibo
├─┬ vite-plugin-ruby@3.1.2
│ └── vite@3.1.0 deduped
└── vite@3.1.0
```
</details>

### Screenshots 📷

_Provide console or browser screenshots of the problem_.

On first load
<img width="1511" alt="image" src="https://user-images.githubusercontent.com/1203658/190829149-798ea0b6-4b59-4d49-9f6d-f5c8e687a9a8.png">

On second load

<img width="1511" alt="image" src="https://user-images.githubusercontent.com/1203658/190829201-83e6b2db-8446-4164-bca5-b0d7a07f8e0b.png">

On manually clearing caches and reloading again
<img width="1512" alt="image" src="https://user-images.githubusercontent.com/1203658/190829244-8140ae1c-a1a5-400b-95e8-87b79805708f.png">

**Comment by ElMassimo** [comment]
Hi Kevin!

Sounds like a bug in Safari, as the spec is very clear on that responses with  `no-cache` should always be revalidated with the server before reusing.

I'll investigate during the week, but given that this library is using Rails `stylesheet_link_tag` it's unlikely that I'll include a workaround for Safari's behavior here.

**Comment by ElMassimo** [comment]
Yeah, adding this to the _Troubleshooting_ section, including the "import from JS" workaround would be nice.

[PRs are welcome](https://github.com/ElMassimo/vite_ruby/blob/main/docs/src/guide/troubleshooting.md)!

**Comment by ElMassimo** [comment]
Closed by #261, thanks!
